src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -100,19 +103,19 @@
             TokenTypes.METHOD_CALL,
             TokenTypes.CLASS_DEF,
             TokenTypes.METHOD_DEF,
-            TokenTypes.LITERAL_IF,
             TokenTypes.LITERAL_FOR,
-            TokenTypes.LITERAL_WHILE,
-            TokenTypes.LITERAL_DO,
```

Comment:
3 tokens removed are because they technically aren't frames.
You can't declare a variable inside an if statement or in an if statement condition.
What starts a new frame is a `{` and not the `if` itself.
Regression shows there are no differences from this action.